src/D2L.CodeStyle.Analyzers/ApiUsage/Logging/LoggingExecutionContextScopeBuilderAnalyzer.cs

```diff
@@ -0,0 +1,147 @@
+			Diagnostics.LoggingContextRunAwaitable
+		);
+
+		public override void Initialize( AnalysisContext context ) {
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction( RegisterScopeBuilderAnalyzer );
+		}
+
+		private void RegisterScopeBuilderAnalyzer( CompilationStartAnalysisContext context ) {
+			var ILoggingExecutionContextScopeBuilder = context.Compilation.GetTypeByMetadataName( "D2L.LP.Logging.ExecutionContexts.ILoggingExecutionContextScopeBuilder" );
+
+			// ILoggingExecutionContextScopeBuilder not in compilation, no need to register
+			if( ILoggingExecutionContextScopeBuilder == null || ILoggingExecutionContextScopeBuilder.Kind == SymbolKind.ErrorType ) {
+				return;
+			}
+
+			ImmutableHashSet<IMethodSymbol> ILoggingExecutionContextScopeBuilderRunSymbols = ILoggingExecutionContextScopeBuilder
+				.GetMembers()
+				.Where( m => m.Kind == SymbolKind.Method )
+				.Where( m => m.MetadataName == "Run" )
+				.Cast<IMethodSymbol>()
+				.ToImmutableHashSet();
+
+			// Can't find ILoggingExecutionContextScopeBuilder.Run or ILoggingExecutionContextScopeBuilder.Run<T>
+			if( !ILoggingExecutionContextScopeBuilderRunSymbols.Any() ) {
+				return;
+			}
+
+			ImmutableHashSet<INamedTypeSymbol> taskTypeBuiltins = ImmutableHashSet
+				.Create(
+					context.Compilation.GetTypeByMetadataName( "System.Threading.Tasks.Task" ),
+					context.Compilation.GetTypeByMetadataName( "System.Threading.Tasks.Task`1" ),
+					context.Compilation.GetTypeByMetadataName( "System.Threading.Tasks.ValueTask" ),
+					context.Compilation.GetTypeByMetadataName( "System.Threading.Tasks.ValueTask`1" )
+				)
+				.Where( x => x != null && x.Kind != SymbolKind.ErrorType )
+				.ToImmutableHashSet();
+
+			INamedTypeSymbol AsyncMethodBuilderAttribute = context.Compilation.GetTypeByMetadataName( "System.Runtime.CompilerServices.AsyncMethodBuilderAttribute" );
+
+			context.RegisterSyntaxNodeAction(
+				ctx => RunInvocationAnalysis(
+					context: ctx,
+					AsyncMethodBuilderAttribute: AsyncMethodBuilderAttribute,
+					ILoggingExecutionContextScopeBuilderRunSymbols: ILoggingExecutionContextScopeBuilderRunSymbols,
+					taskTypeBuiltins: taskTypeBuiltins,
+					invocationSyntax: ctx.Node as InvocationExpressionSyntax
+				),
+				SyntaxKind.InvocationExpression
+			);
+		}
+
+		private void RunInvocationAnalysis(
+			SyntaxNodeAnalysisContext context,
+			IImmutableSet<IMethodSymbol> ILoggingExecutionContextScopeBuilderRunSymbols,
+			INamedTypeSymbol AsyncMethodBuilderAttribute,
+			IImmutableSet<INamedTypeSymbol> taskTypeBuiltins,
+			InvocationExpressionSyntax invocationSyntax
+		) {
+			SemanticModel model = context.SemanticModel;
+
+			SymbolInfo methodSymbolInfo = model.GetSymbolInfo( invocationSyntax, context.CancellationToken );
+
+			bool isInvokingRun = ILoggingExecutionContextScopeBuilderRunSymbols.Contains( methodSymbolInfo.Symbol )
+				|| methodSymbolInfo
+					.CandidateSymbols

```

Comment:
With the spec tests I've ended up in `CandidateSymbols` due to "ambiguous" overloads (they obviously aren't ambiguous in real-life, but appear to be at this phase?).